addon/components/polaris-dropzone.js

```diff
@@ -0,0 +1,522 @@
+import Component from '@ember/component';
+import layout from '../templates/components/polaris-dropzone';
+import { computed } from '@ember/object';
+import { or } from '@ember/object/computed';
+import { capitalize } from '@ember/string';
+import { throttle } from '@ember/runloop';
+import { isNone, isPresent } from '@ember/utils';
+import State from '../-private/dropzone-state';
+import { fileAccepted, getDataTransferFiles } from '../utils/dropzone';
+import ContextBoundEventListenersMixin from 'ember-lifeline/mixins/dom';
+
+const iconDragDrop = 'drag-drop';
+const iconAlertCircle = 'alert-circle';
+
+export default Component.extend(ContextBoundEventListenersMixin, {
+  layout,
+  classNames: ['Polaris-DropZone'],
+  classNameBindings: [
+    'outline:Polaris-DropZone--hasOutline',
+    'isDragging:Polaris-DropZone--isDragging',
+    'state.error:Polaris-DropZone--hasError',
+    'sizeClass',
+  ],
+  attributeBindings: [
+    'ariaDisabled:aria-disabled',
+  ],
+
+  /**
+   * Allowed file types
+   *
+   * @type {String}
+   * @public
+   */
+  accept: null,
+
+  /**
+   *  Sets an active state
+   *
+   * @type {Boolean}
+   * @default false
+   * @public
+   */
+  active: false,
+
+  /**
+   * Allows multiple files to be uploaded
+   *
+   * @type {Boolean}
+   * @default true
+   * @public
+   */
+  allowMultiple: true,
+
+  /**
+   * Sets a disabled state
+   *
+   * @type {Boolean}
+   * @default false
+   * @public
+   */
+  disabled: false,
+
+  /**
+   * Allows a file to be dropped anywhere on the page
+   *
+   * @type {Boolean}
+   * @default false
+   * @public
+   */
+  dropOnPage: false,
+
+  /**
+   * Sets an error state
+   *
+   * @type {Boolean}
+   * @default false
+   * @public
+   */
+  error: false,
+
+  /**
+   * Text that appears in the overlay when set in error state
+   *
+   * @type {String}
+   * @default null
+   * @public
+   */
+  errorOverlayText: null,
+
+  /**
+   * Sets the default file dialog state
+   *
+   * @type {Boolean}
+   * @default false
+   * @public
+   */
+  openFileDialog: false,
+
+  /**
+   *  Displays an outline border
+   *
+   * @type {Boolean}
+   * @default true
+   * @public
+   */
+  outline: true,
+
+  /**
+   * Displays an overlay on hover
+   *
+   * @type {Boolean}
+   * @default true
+   * @public
+   */
+  overlay: true,
+
+  /**
+   * Text that appears in the overlay
+   *
+   * @type {String}
+   * @default null
+   * @public
+   */
+  overlayText: null,
+
+  /**
+   * Whether is a file or an image, `file` | `image`.
+   *
+   * @type {String}
+   * @default 'file'
+   * @public
+   */
+  type: 'file',
+
+  /**
+   * Adds custom validations
+   * (file) => boolean
+   *
+   * @type {Function}
+   * @default no-op
+   * @public
+   */
+  customValidator: null,
+
+  /**
+   * Callback triggered on click
+   * (event) => void
+   *
+   * @type {Function}
+   * @default no-op
+   * @public
+   */
+  onClick: null,
```

Comment:
I think we want `onClick() {}` here

Reply:
in this case we don't...cause there's a default `open` method invoked if this is not present, otherwise we want to invoke the provided `onClick` action (check `handleClick`)...the idea being that if onClick is not provided, by default, it will open the upload select files window, but if provided, it will invoke it and give the user the flexibility to do whatever they want and then open the select files window by setting `openFileDialog` to `true`

Reply:
Ah ok. Well in that case just update the default value since it's listed as a no-op by default.